clang/lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -1010,6 +1010,11 @@ void CudaToolChain::addClangTargetOptions(
     CC1Args.push_back(DriverArgs.MakeArgString(LibSpirvFile));
   }
 
+  bool UseShortPtr = DeviceOffloadingKind == Action::OFK_SYCL;
+  if (DriverArgs.hasFlag(options::OPT_fcuda_short_ptr,
+                         options::OPT_fno_cuda_short_ptr, UseShortPtr))
+    CC1Args.append({"-mllvm", "--nvptx-short-ptr"});
+
```

Comment:
Driver changes look OK - please add associated LIT tests that reflect the new default settings when using `-fsycl`

Reply:
Done, thanks. Let me know if they're acceptable.